### ISSUE TITLE
Browsing history piles up in filter matrix

### ISSUE THREAD

**Issue by chaosvector** [el1t/uBlock-Safari#122]
### Describe the issue

When browsing, any domains of any visited website keep on being listed in the uBlock filter matrix. The filter behaviour degrades by each newly visited websites. Using a fresh tab "resets" this behaviour. 

### One or more specific URLs where the issue occurs

any

### Screenshot in which the issue can be seen

a) The filter matrix after visiting www.bing.com in a fresh browser tab:
![bildschirmfoto 2018-05-26 um 23 05 39](https://user-images.githubusercontent.com/23615285/40580211-a3cf334c-613a-11e8-8a53-eb183a9df36a.png)

b) The filter matrix after visiting www.duckduckgo.com, www.google.com, www.wikipedia.org and then www.bing.com:
![bildschirmfoto 2018-05-26 um 23 05 14](https://user-images.githubusercontent.com/23615285/40580213-a5f009b2-613a-11e8-9355-2d2e587e52c5.png)

### Steps for anyone to reproduce the issue

1. Open an empty tab in Safari
2. Check the filter matrix, no domains listed
3. Visit www.bing.com
4. Check the filter matrix, only bing.com and live.com listed
5. Visit www.duckduckgo.com, www.google.com, www.wikipedia.org
6. Check the filter matrix, any domains of the above websites listed

### Your settings

- MacOS version: 10.13.4
- Safari version: 11.1 (13605.1.33.1.4)
- uBlock Origin version: 1.16.0 (from Safari Extension Gallery)

##### Your filter lists

Checked with and without. No difference.

##### Your custom filters (if any)

none

**Comment by fostermarke** [comment]
This seems to be the same problem as #114

**Comment by chaosvector** [comment]
Yep, this sounds the same as #114. Additionally clicking on the padlock symbol in the filter matrix does not have any effect anymore. Sorry to bring this up a second time.

**Comment by the-real-tokai** [comment]
Yep, it is broken in 1.16.0. I had to revert too (which is a freaking nightmare in Safari as it always wants to send you to the "Gallery" to the latest version when double-clicking the old versions's .safariextz file. A simple TimeMachine restore of the extension seems to be impossible. Entirely broken user experience, IMHO. That's not uBlock's fault, of course, just Apple stupidity.)
